insonmnia/locator/server.go

```diff
@@ -1,183 +1,58 @@
 package locator
 
 import (
-	"crypto/ecdsa"
-	"crypto/tls"
 	"fmt"
-	"net"
-	"sync"
-	"time"
+	"golang.org/x/net/context"
 
 	"github.com/ethereum/go-ethereum/common"
 	log "github.com/noxiouz/zapctx/ctxlog"
-	"github.com/pkg/errors"
-	pb "github.com/sonm-io/core/proto"
-	"github.com/sonm-io/core/util"
 	"go.uber.org/zap"
-	"golang.org/x/net/context"
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/credentials"
-	"google.golang.org/grpc/peer"
-	"google.golang.org/grpc/status"
-)
 
-var errNodeNotFound = errors.New("node with given Eth address cannot be found")
+	ds "github.com/sonm-io/core/insonmnia/locator/datastruct"
+	pb "github.com/sonm-io/core/proto"
+)
 
-type node struct {
-	ethAddr common.Address
-	ipAddr  []string
-	ts      time.Time
+type Storage interface {
+	ByEthAddr(ethAddr common.Address) (*ds.Node, error)
+	Put(n *ds.Node) error
 }
 
-type Locator struct {
-	mx sync.Mutex
+type Srv struct {
+	s Storage
+}
 
-	conf        *LocatorConfig
-	db          map[common.Address]*node
-	ctx         context.Context
-	ethKey      *ecdsa.PrivateKey
-	grpc        *grpc.Server
-	certRotator util.HitlessCertRotator
-	creds       credentials.TransportCredentials
+func NewServer(s Storage) *Srv {
+	return &Srv{s: s}
 }
 
-func (l *Locator) Announce(ctx context.Context, req *pb.AnnounceRequest) (*pb.Empty, error) {
-	ethAddr, err := l.extractEthAddr(ctx)
+func (s *Srv) Announce(ctx context.Context, req *pb.AnnounceRequest) (*pb.Empty, error) {
+	ethAddr, err := ExtractEthAddr(ctx)
 	if err != nil {
 		return nil, err
 	}
 
-	log.G(l.ctx).Info("handling Announce request",
+	log.G(ctx).Info("Handling Announce request",
 		zap.Stringer("eth", ethAddr), zap.Strings("ips", req.IpAddr))
 
-	l.putAnnounce(&node{
-		ethAddr: ethAddr,
-		ipAddr:  req.IpAddr,
+	err = s.s.Put(&ds.Node{
+		EthAddr: ethAddr,
+		IpAddr:  req.IpAddr,
 	})
 
-	return &pb.Empty{}, nil
+	return &pb.Empty{}, err
 }
 
-func (l *Locator) Resolve(ctx context.Context, req *pb.ResolveRequest) (*pb.ResolveReply, error) {
-	log.G(l.ctx).Info("handling Resolve request", zap.String("eth", req.EthAddr))
+func (s *Srv) Resolve(ctx context.Context, req *pb.ResolveRequest) (*pb.ResolveReply, error) {
```

Comment:
"rename everything" strikes back?

Reply:
Could you elaborate on it? You might have noticed that storage/grpc/application concerns have been separated and now reside in their corresponding structures/files.